### PR TITLE
[PM-10294] Remove FC v1 from Clients

### PR DESCRIPTION
**Pull request by vincentsalucci**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10294

## 📔 Objective

> Remove FC v1 references from the clients codebase

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/vault/org-vault/vault.component.ts [hunk 1]
```diff
@@ -855,14 +804,8 @@ export class VaultComponent implements OnInit, OnDestroy {
...
 
-    if (this.flexibleCollectionsV1Enabled) {
-      // V1 limits admins to only adding items to collections they have access to.
-      collections = await firstValueFrom(this.editableCollections$);
-    } else {
-      collections = (await firstValueFrom(this.vaultFilterService.filteredCollections$)).filter(
-        (c) => !c.readOnly && c.id != Unassigned,
-      );
-    }
+    // V1 limits admins to only adding items to collections they have access to.
```
Comment (eliykat) [inline apps/web/src/app/vault/org-vault/vault.component.ts:807]: ⛏️  still some "V1" references in comments in this file.

**Review by eliykat — CHANGES_REQUESTED** [review]
Love this

Screenshot 2024-08-08 at 11 41 48 AM

File: apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts
Comment (shane-melton) [inline apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts:1]: ⛏️ We don't need `ConfigService` anymore in this component 🎉

**Review by shane-melton — CHANGES_REQUESTED** [review]
Looks good! Just one tiny nit for an unused `ConfigService`

**Review by shane-melton — DISMISSED** [review]
Vault changes look good! Thanks for removing all that code, very happy to see it go!

**Review by eliykat — APPROVED** [review]

**Comment by vincentsalucci** [comment]
@shane-melton @eliykat Had to resolve one conflict 😞

**Review by shane-melton — APPROVED** [review]

**Review by eliykat — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Review by eliykat — APPROVED** [review]

**Review by eliykat — DISMISSED** [review]
Please also remove the enum definition.

**Comment by vincentsalucci** [comment]
> Please also remove the enum definition.

Have a follow-up PR ready with this change. Aligns with our Task creation/timing with the server enum.

**Comment by vincentsalucci** [comment]
@bitwarden/dept-design Adding the group to the PR per process: no net-new changes have been made to the UI, rather a checkbox is now visible due to the removal of the Flexible Collections V1 feature flag.

**Review by sukhleenb — APPROVED** [review]

**Pull request merged**